### ISSUE TITLE
Update sections in Configuration > Secrets to remove mentions of Docker

### ISSUE THREAD

**Issue by celestehorgan** [kubernetes/website#30945]
Ref: https://github.com/kubernetes/website/issues/30775

There are three sections that need updating on https://kubernetes.io/docs/concepts/configuration/secret/

- https://kubernetes.io/docs/concepts/configuration/secret/#secret-types
   - specifically, **do we still support docker secret config types?**
- https://kubernetes.io/docs/concepts/configuration/secret/#docker-config-secrets
  - same question as above
- https://kubernetes.io/docs/concepts/configuration/secret/#using-imagepullsecrets
  - Update code samples to no longer use docker

**Comment by Kartik494** [comment]
I would like to give a try on this.Is this going to target dev-1.24 or main?

**Comment by chris-short** [comment]
@chrisnegus and I looked at this one...

> *   specifically, **do we still support docker secret config types?**

This is for Docker Registry config secrets which will still exist in 1.24. No need to touch Secret Types.

> [https://kubernetes.io/docs/concepts/configuration/secret/#docker-config-secrets](https://kubernetes.io/docs/concepts/configuration/secret/#docker-config-secrets)

We'd suggest updating this to "Docker Registry Config Secrets"

> [https://kubernetes.io/docs/concepts/configuration/secret/#using-imagepullsecrets](https://kubernetes.io/docs/concepts/configuration/secret/#using-imagepullsecrets)

We think the wording in here is fine. If we want a code example for a non-Docker registry, what would we choose?  Regardless, I think this section is fine wording wise.

**Comment by Kartik494** [comment]
@chris-short so the final goal is to modify this [docker-config-secrets](https://kubernetes.io/docs/concepts/configuration/secret/#docker-config-secrets) section only? is the example yaml file needs to removed?

**Comment by chrisnegus** [comment]
> @chris-short so the final goal is to modify this [docker-config-secrets](https://kubernetes.io/docs/concepts/configuration/secret/#docker-config-secrets) section only? is the example yaml file needs to removed?

I believe that the recommendation from @chris-short is to only change the section name from "Docker config Secrets" to "Docker Registry config Secrets". The rest of the content is fine because it covers the use of Docker registries and not dockershim.

**Comment by nate-double-u** [comment]
/assign Kartik494
We would be happy for you to work on this!

This can go into `main`.

**Comment by Kartik494** [comment]
Hi , I Have raised a PR . PTAL

**Comment by sftim** [comment]
> This is for Docker Registry config secrets which will still exist in 1.24. No need to touch Secret Types.
>
> https://kubernetes.io/docs/concepts/configuration/secret/#docker-config-secrets
>
> We'd suggest updating this to "Docker Registry Config Secrets"
>  https://kubernetes.io/docs/concepts/configuration/secret/#using-imagepullsecrets
> 
> We think the wording in here is fine. If we want a code example for a non-Docker registry, what would we choose?  Regardless, I think this section is fine wording wise. 

I'm not sure you have that right @chris-short .

The way I have it, the link between the name Docker and these Secret types is that the content of the Secret represents a configuration file. That configuration file follows the same format that you would find on your PC if you used Docker locally.

The page https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/ explains how you can get a file like that onto your PC and then save it into a Kubernetes cluster.

I don't know what different formats of image pull secrets we can pass to a container runtime via CRI, though. I _think_ we only support configuration via Docker-format Secrets and that if we wanted to support something else, https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2133-kubelet-credential-providers is supposed to be the way to go.


So it's `[[Docker (flavored)] [[(registry) config] Secrets]]` not `[[[Docker registry] config] Secrets]`

If we change to writing write “Docker registry” in the heading, readers might think that you only use this kind of Secret when you want to authenticate to Docker Hub.

PS We should (and will) keep these types of Secret, even if the kubelet now only supports CRI integration with container runtimes.

**Comment by Kartik494** [comment]
@sftim Could you please describe what `flavored` registry here means.

**Comment by sftim** [comment]
You can read “Docker (flavored) registry config Secrets” as “Secrets for authenticating to container registries, that follow the format that the `docker` tool uses for configuring registry authentication”.

I think our challenge has been to find a more concise wording that isn't ambiguous.

**Comment by Kartik494** [comment]
> You can read “Docker (flavored) registry config Secrets” as “Secrets for authenticating to container registries, that follow the format that the `docker` tool uses for configuring registry authentication”.

Thanks for the inputs i will take a look on this

**Comment by Kartik494** [comment]
@sftim @celestehorgan as #32092 has been closed , so should we close this issue also?
Thanks!!
